usaspending_api/disaster/v2/views/disaster_base.py

```diff
@@ -1,11 +1,86 @@
 from usaspending_api.submissions.helpers import get_last_closed_submission_date
+from usaspending_api.submissions.models import SubmissionAttributes
+from django.db.models.functions import Concat
+from django.db.models import Value, CharField
+
+COVID_19_GROUP_NAME = "covid_19"
+
+
+def covid_def_codes():
+    return DisasterEmergencyFundCode.objects.filter(group_name=COVID_19_GROUP_NAME)
+
+
+def covid_def_code_strings():
+    return [code["code"] for code in covid_def_codes().values("code")]
+
+
+def latest_gtas_of_each_year_queryset():
+    return (
+        GTASSF133Balances.objects.annotate(
+            is_highest_period=Exists(
+                GTASSF133Balances.objects.values("fiscal_year")
+                .annotate(
+                    fiscal_period_max=Max("fiscal_period"),
+                    include=Exists(
+                        DABSSubmissionWindowSchedule.objects.filter(
+                            is_quarter=False,
+                            submission_fiscal_year=OuterRef("fiscal_year"),
+                            submission_fiscal_month=OuterRef("fiscal_period"),
+                            submission_reveal_date__lte=datetime.now(timezone.utc),
+                        )
+                    ),
+                )
+                .filter(
+                    include=True,
+                    fiscal_year=OuterRef("fiscal_year"),
+                    fiscal_period_max=OuterRef("fiscal_period"),
+                    disaster_emergency_fund_code__in=covid_def_code_strings(),
+                )
+                .values("fiscal_year", "fiscal_period_max")
+            )
+        )
+        .annotate(fiscal_period_and_year=Concat("fiscal_year", Value("/"), "fiscal_period", output_field=CharField()))
+        .filter(
+            is_highest_period=True,
+            fiscal_period_and_year__in=[
+                f"{elem['submission_fiscal_year']}/{elem['submission_fiscal_month']}"
+                for elem in DABSSubmissionWindowSchedule.objects.filter(is_quarter=False)
+                .order_by("-submission_fiscal_year", "-submission_fiscal_quarter", "-submission_fiscal_month")
+                .values()
+            ],
+        )
+    )
+
+
+def latest_faba_of_each_year_queryset():
+    return FinancialAccountsByAwards.objects.filter(submission__in=latest_submission_of_each_year_queryset())
+
+
+def latest_submission_of_each_year_queryset():
+    return SubmissionAttributes.objects.annotate(
+        include=Exists(
+            SubmissionAttributes.objects.values("reporting_fiscal_year")
+            .annotate(fiscal_period_max=Max("reporting_fiscal_period"))
+            .values("reporting_fiscal_year", "fiscal_period_max")
+            .filter(
+                reporting_fiscal_year=OuterRef("reporting_fiscal_year"),
+                reporting_fiscal_year__gte=2020,
+                fiscal_period_max=OuterRef("reporting_fiscal_period"),
```

Comment:
Further, in this case we _do_ need to factor in `DABSSubmissionWindowSchedule.is_quarter` as part of the uniqueness/group that is returned. This is because we need to grab the latest closed month AND latest closed quarter. 

At end of FYs, e.g. end of FY2020 in nov/dec ... that would be:
1. `submission_fiscal_year=2020, submission_fiscal_quarter=4, submission_fiscal_month=12, is_quarter=True`
1. `submission_fiscal_year=2020, submission_fiscal_quarter=4, submission_fiscal_month=12, is_quarter=False`

Which are two different submission windows/periods. 

But Around Aug 5, 2020, before Q3 has closed by after P 7-9 has closed, it should return:
1. `submission_fiscal_year=2020, submission_fiscal_quarter=2, submission_fiscal_month=6, is_quarter=True`
1. `submission_fiscal_year=2020, submission_fiscal_quarter=3, submission_fiscal_month=9, is_quarter=False`

Reply:
This is a corner-case btw, and grabbing only the latest submitted _month_ should cover the vast majority of data that we are after. But it is possible that agencies will still submit COVID data in _quarterly_ submissions -- Broker has nothing in place yet to prevent this.